### PR TITLE
[rayci] add macos sandbox plugin

### PR DESCRIPTION
**Pull request by can-anyscale**
Add macos sandbox plugin for civ2 (similar to civ1: https://github.com/ray-project/rayci/blob/main/ray_ci/step_macos.json#L4)

Test:
- CI

### CODE REVIEW

**Comment by can-anyscale** [comment]
Huh included weird changes, hold on

File: raycicmd/converter.go [hunk 1]
```diff
@@ -260,6 +262,12 @@ func (c *converter) convertRunner(step map[string]any) (map[string]any, error) {
 		result["plugins"] = []any{map[string]any{
 			dockerPlugin: makeRayWindowsDockerPlugin(dockerPluginConfig),
 		}}
+	} else if jobEnv == macosJobEnv { // a special job env for macos
+		result["plugins"] = []any{map[string]any{
+			macosSandboxPlugin: map[string]string{
+				"deny-file-read": "/usr/local/etc/buildkite-agent/buildkite-agent.cfg",
+			},
+		}}
```
Comment (aslonnie) [inline raycicmd/converter.go:270]: could you declare a global var for this `map` value?

one goal is that we do not want to repeat this value in the tests.

File: raycicmd/converter_test.go [hunk 2]
```diff
@@ -303,9 +334,15 @@ func TestConvertPipelineStep(t *testing.T) {
 		_, isWait := got["wait"]
 		_, isWanda := test.in["wanda"]
 
-		plugins, ok := got["plugins"]
+		isMac := false
+		jobEnv, ok := test.in["job_env"]
 		if ok {
-			// Check non plugins only.
+			isMac = jobEnv == macosJobEnv
+		}
```
Comment (aslonnie) [inline raycicmd/converter_test.go:341]: nit: just write:

`isMac := ok && jobEnv == macosJobEnv` ?

you can even ignore the `ok` part, as `jobEnv` will be empty string when it is not `ok`, basically:


```
jobEnv, _ := test.in["job_env"]
isMac := jobEnv == macosJobEnv
```

File: raycicmd/converter_test.go [hunk 3]
```diff
@@ -260,6 +261,36 @@ func TestConvertPipelineStep(t *testing.T) {
...
+				"RAYCI_BRANCH":                          "beta",
+				"RAYCI_BUILD_ID":                        "abc123",
+				"RAYCI_TEMP":                            "s3://ci-temp/abc123/",
+				"RAYCI_WORK_REPO":                       "fakeecr",
+			},
+			"plugins": []any{map[string]any{
+				macosSandboxPlugin: map[string]string{
+					"deny-file-read": "/usr/local/etc/buildkite-agent/buildkite-agent.cfg",
+				},
+			}},
```
Comment (aslonnie) [inline raycicmd/converter_test.go:292]: please remove this and check plugin from the extracted `plugins`. to be consistent with the semantics of other `out` values.

**Review by aslonnie — APPROVED** [review]

File: raycicmd/converter.go [hunk 4]
```diff
@@ -8,6 +8,8 @@ import (
 )
 
 const windowsJobEnv = "WINDOWS"
+const macosJobEnv = "MACOS"
+const macosDenyFileRead = "/usr/local/etc/buildkite-agent/buildkite-agent.cfg"
```
Comment (aslonnie) [inline raycicmd/converter.go:12]: what I meant is that to define a:

````
var macosPlugin = []any{map[string]any{
	macosSandboxPlugin: map[string]string{
		"deny-file-read": macosDenyFileRead,
	},
}
````

but I guess this is also fine.

Comment (can-anyscale) [reply raycicmd/converter.go:12]: w00t already merged

**Pull request merged**